File: atcoder/abs/9.pas
program ABC049C;
uses StrUtils;
var
	T: array[0..3] of string = ('dream','dreamer','erase','eraser');
	s: string;
	i,c,l,k: integer;
begin
	for i:=0 to 3 do begin
		T[i]:=ReverseString(T[i]);
	end;
	readln(s);
	s:=ReverseString(s);
	c:=0;
	l:=Length(s);
	while c<l do begin
		k:=-1;
		for i:=0 to 3 do begin
			if copy(s,c+1,Length(T[i]))=T[i] then begin
				k:=Length(T[i]);
			end;
		end;
		if k<0 then begin
			writeln('NO');
			exit;
		end;
		c:=c+k;
	end;
	writeln('YES');
end.

File: atcoder/abs/9.vb
module ABC049C
	sub Main(ByVal args() as String)
		dim T() as string = {"dream","dreamer","erase","eraser"}
		dim s as string
		dim i,c,l,k as integer
		for i=0 to 3
			T(i)=StrReverse(T(i))
		next
		s=StrReverse(Console.ReadLine())
		c=0
		l=s.Length
		while c<l
			k=-1
			for i=0 to 3
				if l-c>=T(i).Length andalso s.Substring(c,T(i).Length)=T(i)
					k=T(i).Length
					exit for
				end if
			next
			if k<0
				Console.WriteLine("NO")
				return
			end if
			c+=k
		end while
		Console.WriteLine("YES")
	end sub
end module
